mlflow/pyfunc/__init__.py

```diff
@@ -409,23 +412,34 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
                      (i.e. read / write the elements using C-like index order), and DataFrame
                      column values will be cast as the required tensor spec type.
 
+        :param parameters: Additional parameters to pass to the model for inference.
+
         :return: Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
         input_schema = self.metadata.get_input_schema()
         if input_schema is not None:
             data = _enforce_schema(data, input_schema)
 
+        if parameters is not None:
+            if not isinstance(parameters, dict):
+                raise MlflowException(
+                    "Parameters must be a dictionary. Got type '{}'.".format(type(parameters))
+                )
+            parameters_schema = self.metadata.get_parameters_schema()
+            if parameters_schema is not None:
```

Comment:
Do we want to always validate this or only validate when it's set? @dbczumar

Reply:
I think that's the only way to validate the non-optional schema definitions, right? 
To make models.utils:889 reachable for enforcement this has to execute.